src/Validation/JpValidation.php

```diff
@@ -25,13 +25,46 @@ class JpValidation extends LocalizedValidation
 {
     /**
      * Checks a phone number for Japan.
+     * Accepts the following format.
+     * - 0311112222
+     * - 03 1111 2222 or 09876 1 2222
+     * - 03-1111-2222 or 09876-1-2222
+     * - +81311112222
+     * - +81 3 1111 2222
+     * - +81-3-1111-2222
+     * - 09011112222
+     * - 090 1111 2222
+     * - 090-1111-2222
+     * - +819011112222
+     * - +81 90 1111 2222
+     * - +81-90-1111-2222
+     * - 0120111222
+     * - 0120 111 222
+     * - 0120-111-222
      *
      * @param string $check The value to check.
      * @return bool Success.
      */
     public static function phone($check)
     {
-        $pattern = '/^(0\d{1,4}[\s-]?\d{1,4}[\s-]?\d{1,4}|\+\d{1,3}[\s-]?\d{1,4}[\s-]?\d{1,4}[\s-]?\d{1,4})$/';
+        $pattern = '/^(?:(?:0|\+81[\s\-]?)[1-9](?:0(?:\d{8}|(?:[\s\-]\d{4}){2})|(?:[1-9]\d{7}|(?:[1-9]0(?:[\s\-]\d{3}){2}|\d{0,3}[\s\-]\d{1,4}[\s\-]\d{4}))))$/';
```

Comment:
<pre>
/^
  (?:
    (?: 0 | \+81[\s\-]? )[1-9]
    (?:
      0
      (?:
        \d{8}
      | (?:[\s\-]\d{4}){2}
      )
    |
      (?:
        [1-9]\d{7}
      | 
        (?:
          [1-9] 0 (?:[\s\-]\d{3}){2}
        | <strong>\d{0,3}</strong> [\s\-] \d{1,4} [\s\-] \d{4}
        )
      )
    )
  )
$/x
</pre>

Sorry if it is intentional, but `\d{0,3}` marked as bold seems incorrect to me. Wouldn't it return true against <code>01<strong>000</strong>-0000-0000</code>? This format is not listed in the doc comment.